### DISCUSSION TITLE
Error while trying to stream a RTSP to Kinesis Video Stream using kvssink

### DISCUSSION THREAD

**Discussion by vbbandeira** [bluenviron/mediamtx#690] [Questions]
Hello there folks,

I'm trying to stream an RTSP to the AWS Kinesis Video Streams, based on your service. I was able to set up the stream, watch it running well in VLC, and so on.

Then, I've started to follow the tutorial on [https://github.com/aws-samples/amazon-kinesis-video-streams-demos/tree/master/producer-cpp/docker-rtsp](https://github.com/aws-samples/amazon-kinesis-video-streams-demos/tree/master/producer-cpp/docker-rtsp), and after running the ```kvs_gstreamer_sample```, I got an error in docker:

```
0:00:02.597636000  1035 0x55cfcc491320 INFO        GST_ERROR_SYSTEM gstelement.c:2145:gst_element_message_full_with_details:<source> posting message: Internal data stream error.
Error received from element source: Internal data stream error.
Debugging information: gstrtspsrc.c(5653): gst_rtspsrc_loop (): /GstPipeline:rtsp-kinesis-pipeline/GstRTSPSrc:source:
streaming stopped, reason not-negotiated (-4)
```

When looking at the container log running rtsp-simple-server, I see the following: it can be found at [https://gist.github.com/vbbandeira/2ba03f7b5aa48b972864288554753f42](https://gist.github.com/vbbandeira/2ba03f7b5aa48b972864288554753f42)

Note: When I try to make the process to AWS passing a remote rtsp the error above doesn't happen, it appears only when I pass the rtsp aler9

Can you guys please lemme know what is going on there?

Thanks!

**Comment by aler9** [comment]
I finally got the change to try AWS Kinesis and stream from rtsp-simple-server to Kinesis, and in my case it worked flawlessly. Here are the steps i followed:

1. in one window, launch rtsp-simple-server

   ```
   docker run --rm -it --network=host aler9/rtsp-simple-server
   ```

2. in another window, publish a file to rtsp-simple-server

   ```
   ffmpeg -re -stream_loop -1 -i sample_1080p_h264.mp4 -c copy -f rtsp rtsp://localhost:8554/mystream
   ```

3. Log into AWS, click the user menu, "My Security Credentials", "Access keys", "Create access key", copy key id and access key

4. launch `kvs_gstreamer_sample` with the key ID and access key:

   ```
   git clone https://github.com/aws-samples/amazon-kinesis-video-streams-demos
   cd amazon-kinesis-video-streams-demos/producer-cpp/docker-rtsp
   docker build -t rtspdockertest .
   docker run -it --network=host rtspdockertest:latest /bin/bash
   AWS_ACCESS_KEY_ID=mykey AWS_SECRET_ACCESS_KEY=myaccess ./kvs_gstreamer_sample teststream rtsp://localhost:8554/mystream
   ```

Just replicate the above and it will work.

[a comment by an automated account is not shown]
